src/java/org/apache/cassandra/utils/RMICloseableServerSocketFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.io.IOException;
+import java.rmi.server.RMIServerSocketFactory;
+
+/**
+ * This represents closeable RMI Server Socket factory.
+ */
+public interface RMICloseableServerSocketFactory extends RMIServerSocketFactory
```

Comment:
This is essentially:
```suggestion
public interface RMICloseableServerSocketFactory extends RMIServerSocketFactory, AutoCloseable
```
I am trying to figure out if there's a reason we _wouldn't_ want to also extend `AutoCloseable` rather than adding a custom `close()` method... while I can't think of a reason why you'd want to use one in a "try-with-resources" block at this time, it also doesn't hurt anything to _allow it_ in the future, and by not doing it now we're artificially limiting the ability to do so. Also, there are some frameworks that can check if something is AutoCloseable and close it on shutdown/cleanup (again, nothing today seems to do this but it's possible we may want to some day).

Reply:
FYI - this would go for the Client socket factory as well.

Reply:
Makes sense. Will do the changes. Since I've to remove the `close()` method in this case, will add to my changes instead of picking up the commit suggestion.